scene/gui/button.cpp

```diff
@@ -342,16 +340,21 @@ Size2 Button::get_minimum_size_for_text_and_icon(const String &p_text, Ref<Textu
 		minsize.width = 0;
 	}
 
-	if (!expand_icon && !p_icon.is_null()) {
-		minsize.height = MAX(minsize.height, p_icon->get_height());
+	if (!p_icon.is_null()) {
+		float _icon_width_cached = 0;
+
+		if (!expand_icon) {
+			minsize.height = MAX(minsize.height, p_icon->get_height());
+			_icon_width_cached = p_icon->get_width();
+		}
 
 		if (icon_alignment != HORIZONTAL_ALIGNMENT_CENTER) {
-			minsize.width += p_icon->get_width();
+			minsize.width += _icon_width_cached;
 			if (!xl_text.is_empty() || !p_text.is_empty()) {
-				minsize.width += get_theme_constant(SNAME("hseparation"));
+				minsize.width += MAX(0, get_theme_constant(SNAME("h_separation")));
```

Comment:
This change makes it impossible to have negative `h_separation`. I don't think it's necessary to limit it (or at least the limitation should be documented).

Reply:
A negative value of `h_separation` is meaningless, otherwise, it may make `minsize.width` smaller, become `0` or even a **negative** value (even if there is an icon or text). In some cases this can result in insufficient visible space for drawing.

Reply:
It makes the text overlap icon. Weird, but maybe someone wants it.
Limit it is fine too, I guess. But as I said, it should be documented in the property description.